metricbeat/module/kafka/partition/partition.go

```diff
@@ -322,3 +320,111 @@ func checkRetryQuery(err error) (retry, reconnect bool) {
 
 	return false, false
 }
+
+func findMatchingBroker(
+	addr string,
+	advertised string,
+	brokers []*sarama.Broker,
+) *sarama.Broker {
+	// compare connection address to list of broker addresses
+	for _, b := range brokers {
+		if advertised != "" && b.Addr() == advertised {
+			return b
+		}
+
+		if b.Addr() == addr {
```

Comment:
The address returned by `b.Addr()` is the advertised hostname by current broker in metadata (b is not the broker we connect to, but a broker in meta.Brokers).

The advertised hostname must not match the hostname configured. e.g. kafka broker advertises hostname `kafkaX.mydomain.com:9092` and addr could be `localhost:9092`